presentation/src/main/java/com/architect/coders/mu8/splash/SplashViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.architect.coders.mu8.splash
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.architect.coders.mu8.BuildConfig
+import com.architect.coders.mu8.utils.Scope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class SplashViewModel : ViewModel(), Scope by Scope.Impl() {
+
+    companion object {
+        const val TIME_SLEEP: Long = 5000
+    }
+    sealed class UiModel {
+        class GetVersion(val versionName: String) : UiModel()
+        class Navigation : UiModel()
+    }
+
+    private val _model = MutableLiveData<UiModel>()
+    val model: LiveData<UiModel>
+        get() {
+            if (_model.value == null) onGetVersion()
+            return _model
+        }
+
+    init {
+        initScope()
+    }
+
+    private fun onGetVersion() {
+        _model.value = UiModel.GetVersion(BuildConfig.VERSION_NAME)
+        launch {
+            delay(TIME_SLEEP)
+            onNext()
```

Comment:
Rename this method, think of a name that explains what do you want to do next.

Reply:
change name onChangeOfNavigationStatus